### DISCUSSION TITLE
Short title YAML option for Beamer

### DISCUSSION THREAD

**Discussion by wklimowicz** [quarto-dev/quarto-cli#5620] [Feature Requests]
In some beamer themes, the title of the presentation shows up at the bottom of each slide. Some titles don't fit:

![image](https://github.com/quarto-dev/quarto-cli/assets/62679873/afe4e788-7c88-4b31-898f-b6e79e33ccba)

There is an option for adding a shorter title at the bottom with square brackets on `\title`:

```tex
\title[Short Title]{This is the long title}
```
This works for beamer, but also for some document classes like `amsart`.

---

This seems like a clean implementation (line breaks for clarity):

[src/resources/formats/pdf/pandoc/title.tex](https://github.com/quarto-dev/quarto-cli/blob/387e489f344a6a36e8f3ec0be4adb02fd0ddadd1/src/resources/formats/pdf/pandoc/title.tex#LL2C9-L2C9):

```diff
\title
+$if(shorttitle)$[$shorttitle$]$endif$
{$title$$if(thanks)$\thanks{$thanks$}$endif$}
```

And the `qmd` file:

```yaml
---
title: Long title
format:
  beamer:
    shorttitle: Short title
---
``` 

An easy PR but I thought I'd check here first -- any thoughts? It would only work for some LaTeX formats, and I don't know whether the YAML key should be `short-title` or `shorttitle`.

**Comment by mcanouil** [comment]
It has to be in an if Beamer statement, because the partial is not specific to Beamer.

See <https://quarto.org/docs/reference/formats/pdf.html> for other options.
For short, words are delimited by a dash.

Feel free to submit a PR.

**Reply by wklimowicz** [reply]
Actually, looks like this is already called `title-short` in the ACS template: https://github.com/quarto-journals/acs/blame/main/template.qmd -- probably best to keep using that one.

In terms of the if statement, this works for some other formats aside from beamer. For example, using `amsart` with the same implementation as above:

````md
---
title: This is a very long title that runs off to the side
title-short: Short title
format: 
  pdf:
    documentclass: amsart
---
# Foo
\pagebreak
# Bar
````
Page 1:
![image](https://github.com/quarto-dev/quarto-cli/assets/62679873/c4dc65fa-55fd-44c9-830a-349e8c4e6deb)

Page 2:
![image](https://github.com/quarto-dev/quarto-cli/assets/62679873/30d09d31-b35a-4174-8ca3-67af4b353565)

Would this justify keeping it as a more general partial?

**Reply by mcanouil** [reply]
The real question is why this was not in Pandoc's template, if it is an option used by "most" classes.

**Reply by Mgonrub** [reply]
I'm not sure what you're asking, but I'm learning to use quarto (and with jupyter notebook) and I want to apply with beamer.
I am using themes in beamer. And I use these 2 options in the YAML:
`title: long title. 
 `
`subtitle: and the short title`

**Reply by mcanouil** [reply]
@Mgonrub are you sure you are answering the right thread?

**Comment by Mgonrub** [comment]
Hi,
I offer my humble apologies, I was just trying to help. Saying that there was another option to put titles. Here I leave an example.
https://github.com/Mgonrub/beamer1/blob/main/z2.pdf
De nuevo, lo siento mucho (Again, sorry so much...)...

**Reply by mcanouil** [reply]
The goal is to have the short title feature from LaTeX, not title or subtitle, _i.e._, no need to shorten the title everywhere, the short title will appear at the bottom while you'll have the actual and possibly long title as title on the title slide.

PS: @Mgonrub no need to apologise and thank you anyway for the suggested "trick".

**Comment by raffaelemancuso** [comment]
@wklimowicz Was this implemented at last?

**Reply by wklimowicz** [reply]
There's an open PR in #5627. Your hack below is probably easiest for the time being.

**Comment by raffaelemancuso** [comment]
Here is a hack altough it's horrible:

```
title: "Placeholder, otherwise the title page would not be rendered."
header-includes:
  - \AtBeginDocument{\title[ShortTitle]{LongTitle}}
  - \AtBeginDocument{\subtitle[ShortSubTitle]{LongSubTitle}}
  - \AtBeginDocument{\author[ShortAuthor]{LongAuthor}}
  - \AtBeginDocument{\institute[ShortInstitute]{LongInstitute}}
  - \usepackage[italian]{datetime2}
  - \newcommand{\mydate}{\DTMdisplaydate{2024}{10}{09}{-1}}
  - \AtBeginDocument{\date[\DTMsetdatestyle{iso}\mydate]{\mydate}}
```

[Credit](https://stackoverflow.com/questions/50225802/r-markdown-pandoc-beamer-presentation-with-short-title)

EDIT: Changed the hack to also include the other entries

@mcanouil can we move this into an issue?

**Comment by raffaelemancuso** [comment]
This was fixed upstream: https://github.com/jgm/pandoc/issues/10248

@mcanouil Do we need to do something for it to propagate here? Or is it automatic as a new quarto version will include a pandoc that support this?

**Reply by cscheid** [reply]
We've already updated Pandoc for 1.6, and we only update Pandoc once a release. Quarto 1.7 will include this.

**Reply by mcanouil** [reply]
Please don't ping for no reason. Thank you!

Embedded version of Pandoc will be updated when it will be updated. And this will only happen after Pandoc has made a release and after testing in Quarto which happens once or twice within a release development.
